tests/unit/test_data.py

```diff
@@ -160,6 +173,53 @@ def test_normalize(self, data_type, weights):
             normalized_with_zeros.weights, jnp.nan_to_num(expected_weights)
         )
 
+    def test_jaxtyping_compatibility(self, data_type):
+        """
+        Test `Data` compatibility with jaxtyping annotations.
+
+        Checks the following cases:
+            - Correct shape and data type,
+            - Correct shape and incorrect data type,
+            - Incorrect shape and correct data type.
```

Comment:
Could maybe also parametrise the shape we check as well, but that might be less clear than doing two separate checks

Reply:
I think it is clearer to do these checks separately.